spec/services/wallets/recurring_transaction_rules/validate_service_spec.rb

```diff
@@ -17,6 +17,20 @@
       expect(validate_service.call).to be_truthy
     end
 
+    context "when invalid metadata" do
+      let(:params) do
+        {
+          trigger: "interval",
+          interval: "weekly",
+          metadata: {'key' => {'nested_key' => 'value'}}
+        }
+      end
+
+      it "returns false" do
+        expect(validate_service.call).to be_falsey
```

Comment:
`be_falsey` will match with nil and false... as this service returns only true or false I would set the expectation to false instead of falsey...

```suggestion
        expect(validate_service.call).to eq false
```